src/Illuminate/Validation/Rules/Email.php

```diff
@@ -0,0 +1,339 @@
+    protected $customRules = [];
+
+    /**
+     * The error message after validation, if any.
+     *
+     * @var array
+     */
+    protected $messages = [];
+
+    /**
+     * The data under validation.
+     *
+     * @var array
+     */
+    protected $data;
+
+    /**
+     * The validator performing the validation.
+     *
+     * @var \Illuminate\Validation\Validator
+     */
+    protected $validator;
+
+    /**
+     * The callback that will generate the "default" version of the file rule.
+     *
+     * @var string|array|callable|null
+     */
+    public static $defaultCallback;
+
+    /**
+     * Set the default callback to be used for determining the email default rules.
+     *
+     * If no arguments are passed, the default email rule configuration will be returned.
+     *
+     * @param  static|callable|null  $callback
+     * @return static|void
+     */
+    public static function defaults($callback = null)
+    {
+        if (is_null($callback)) {
+            return static::default();
+        }
+
+        if (! is_callable($callback) && ! $callback instanceof static) {
+            throw new InvalidArgumentException('The given callback should be callable or an instance of '.static::class);
+        }
+
+        static::$defaultCallback = $callback;
+    }
+
+    /**
+     * Get the default configuration of the file rule.
+     *
+     * @return static
+     */
+    public static function default()
+    {
+        $email = is_callable(static::$defaultCallback)
+            ? call_user_func(static::$defaultCallback)
+            : static::$defaultCallback;
+
+        return $email instanceof self ? $email : new self();
+    }
+
+    /**
+     * A strict rule set which includes DNS and Spoof checks.
+     *
+     * @return static
+     */
+    public static function strictSecurity()
+    {
+        return (new self())->strict()->dns()->spoof();
+    }
+
+    /**
+     * Validate against NoRFCWarningsValidation.
+     *
+     * @return $this
+     */
+    public function strict()
+    {
+        $this->strict = true;
+
+        return $this;
+    }
+
+    /**
+     * Validate against DNSCheckValidation.
+     * Requires the PHP intl extension.
+     *
+     * @return $this
+     */
+    public function dns()
+    {
+        $this->dns = true;
+
+        return $this;
+    }
+
+    /**
+     * Validate against SpoofCheckValidation.
+     * Requires the PHP intl extension.
+     *
+     * @return $this
+     */
+    public function spoof()
+    {
+        $this->spoof = true;
+
+        return $this;
+    }
+
+    /**
+     * Validate against FilterEmailValidation.
+     *
+     * @return $this
+     */
+    public function filter()
+    {
+        $this->filter = true;
+
+        return $this;
+    }
+
+    /**
+     * Validate against FilterEmailValidation::unicode().
+     *
+     * @return $this
+     */
+    public function filterUnicode()
+    {
+        $this->filter_unicode = true;
+
+        return $this;
+    }
```

Comment:
Not sure if `unicode` should just be a flag for `filter()`:
```suggestion
    /**
     * Validate against FilterEmailValidation.
     *
     * @return $this
     */
    public function filter(bool $unicode = false)
    {
        if ($unicode) {
            $this->filter_unicode = true;
        } else {        
            $this->filter = true;
        }

        return $this;
    }
```

Reply:
Of course, `filterUnicode()` could optionally stay as an alias:
```suggestion
    /**
     * Validate against FilterEmailValidation.
     *
     * @return $this
     */
    public function filter(bool $unicode = false)
    {
        if ($unicode) {
            $this->filter_unicode = true;
        } else {        
            $this->filter = true;
        }

        return $this;
    }

    /**
     * Validate against FilterEmailValidation::unicode().
     *
     * @return $this
     */
    public function filterUnicode()
    {
        return $this->filter(true);
    }
```

Reply:
Interestingly though, there's already something similar in form of [`\Illuminate\Validation\Concerns\FilterEmailValidation`](https://github.com/laravel/framework/blob/11.x/src/Illuminate/Validation/Concerns/FilterEmailValidation.php):

https://github.com/laravel/framework/blob/8b93d39c4721a0c170b92bfbf7bfb5090b5407cc/src/Illuminate/Validation/Concerns/FilterEmailValidation.php#L29-L37

Reply:
I kept the code as close as possible as the current `File` implementation. I will leave it up to Taylor to decide if he wants to follow your suggestions or not

Reply:
I actually don't know if he even reads other people's reviews 🤔 At least, I've never seen him reacting to any.